From f8a2f64b1b485c31cc2d12b805fdb23a87d3e26f Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Tue, 28 Dec 2021 18:29:26 +0530
Subject: [PATCH] ath11k: disable spectral scan during spectral deinit

Fix target assert during rmmod with spectral scan enabled.

Disable spectral scan during spectral deinit if spectral scan is
enabled.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/spectral.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -8068,6 +8068,11 @@ static void ath11k_mac_op_remove_interfa
 
 	mutex_lock(&ar->conf_mutex);
 
+	ret = ath11k_spectral_vif_stop(arvif);
+	if (ret)
+		ath11k_warn(ab, "failed to stop spectral for vdev %i: %d\n",
+			    arvif->vdev_id, ret);
+
 	if (vif->type == NL80211_IFTYPE_AP_VLAN) {
 		ath11k_nss_ext_vdev_delete(arvif);
 
